src/components/AnalysisPage/GeoAnalysisBox/GeoAnalysisBox.jsx

```diff
@@ -36,28 +28,36 @@ const GeoAnalysisBox = ({ partId }) => {
       setSurfaceArea(geoData.surface_area);
       setNumBoundaryEdges(geoData.num_boundary_edges);
       setNumShells(geoData.num_shells);
-      setRoundedVolume(Math.round(geoData.volume));
-      setRoundedSurfaceArea(Math.round(geoData.surface_area));
-      setRoundedNumBoundaryEdges(Math.round(geoData.num_boundary_edges));
-      setRoundedNumShells(Math.round(geoData.num_shells));
     }
   }, [geoData]);
 
+  const displayValue = (value) => {
+    if (typeof value !== "number") return value;
```

Comment:
I found the display code a bit hard to understand because of all the type checking between numbers and strings and stuff. I have some ideas for making it simpler to understand. 

- Move the displayValue function into the GeoStatistic component rather than in the GeoAnalysisBox component.
- Start the values at null again instead of as '--'
- So now the props can just be 'number' type rather than number OR string
- In the displayValue function convert null -> '--'

So now there doesn't need to be all these checks for string or number and the GeomStatistic component is the only place that uses displayValue so it makes sense for that logic to go in there

Let me know if Im missing anything or if you were checking the typeof for a reason other than just looking for the '--'.

Reply:
I'll give this a try!!!